src/tophu/multiscale.py

```diff
@@ -0,0 +1,482 @@
+import warnings
+from typing import Optional, Tuple
+
+import numpy as np
+import scipy.signal
+from numpy.typing import ArrayLike, NDArray
+
+from .filter import bandpass_equiripple_filter
+from .multilook import multilook
+from .tile import TiledPartition
+from .unwrap import UnwrapCallback
+from .upsample import upsample_nearest
+
+__all__ = [
+    "multiscale_unwrap",
+]
+
+
+def lowpass_filter_and_multilook(
+    arr: ArrayLike,
+    downsample_factor: Tuple[int, int],
+    *,
+    shape_factor: float = 1.5,
+    overhang: float = 0.5,
+    ripple: float = 0.01,
+    attenuation: float = 40.0,
+) -> NDArray:
+    r"""
+    Apply an anti-aliasing pre-filter, then multilook.
```

Comment:
Ok after this comment I looked at the multilook function and I noticed that it actually does average. Basically from the higher resolution to the low resolution, it looks like we are doing two filters and one decimation (note that I consider multilook itself as a low-pass filter with average kernel plus decimation). 
I think it is a great functionality built in and extra smoothing usually helps with unwrapping. However, is it configurable if the user wants to apply only multi-looking and avoid the extra filtering?
That would give us the opportunity to understand the impact of the extra filter.

Reply:
Thanks for the review @hfattahi!

You're correct -- the "multilooking" step does perform spatial averaging. That step is functionally equivalent to applying a uniform weight filter and then decimating, so overall this `lowpass_filter_and_multilook()` function is effectively filtering twice before decimating.

My reasoning for filtering twice is that the two filters are intended to accomplish two slightly different things. The initial low-pass filter is there to reduce aliasing effects, since the input interferogram may contain high-frequency content that will otherwise cause artifacts when the interferogram is downsampled. The goal of this filter is to suppress the high-frequency content that would be undersampled after decimating, while preserving the low-frequency components. The second (spatial averaging) filter is intended to improve SNR by taking the mean of nearby pixels.

There is some overlap in the functionality of the two filters -- the low-pass filter does beat down the noise power, and the spatial averaging filter does suppress the higher frequencies a bit. But I don't think that spatial averaging alone would be sufficient to avoid aliasing effects that would distort the phase gradients in the downsampled interferogram.

The figures below help demonstrate this point. They show the frequency response of each filter for different downsampling factors. The dashed vertical line indicates the where the new Nyquist rate would be after downsampling.

![figure1](https://user-images.githubusercontent.com/12984092/189554967-28e693d7-42a1-41fb-b779-67881340bf81.png)
![figure2](https://user-images.githubusercontent.com/12984092/189554976-9e277d5f-7288-4844-93ce-9404e1a7b6ba.png)

[Notebook](https://github.com/gmgunter/tophu-notebooks/blob/main/LowPass-vs-Spatial-Average-Response.ipynb)

>is it configurable if the user wants to apply only multi-looking and avoid the extra filtering?

Right now, the low-pass filter is always applied whenever the interferogram is multilooked. I'll add an option to let the user disable it. Should be a pretty trivial change.

Reply:
Thank you @gmgunter for the nice plots and detailed explanation. It makes sense and nicely shows how the low-pass filter suppresses frequencies higher than nyquist.
It will be really great to make the filter configurable so that we could evaluate the impact on simulated and real data.